src/Form/SettingsForm.php

```diff
@@ -67,6 +89,8 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
       ->set('api.customer_number', $form_state->getValue('api_customer_number'))
       ->set('api.username', $form_state->getValue('api_username'))
       ->set('api.password', $form_state->getValue('api_password'))
+      ->set('api.mode', $form_state->getValue('api_mode'))
+      ->set('shipping_info.origin_postal_code', $form_state->getValue('origin_postal_code'))
```

Comment:
On the schema file you have this as 'api.origin_postal_code'. Here you save it as `shipping_info.origin_postal_code`. This setting is specific to the API so I think it should go under `api.*`. We may put other generic module settings in the future outside of it. `shipping_info` is too generic, it doesn't mean anything. So I think it should be saved under a section specific to the Rate API as `api.rate.origin_postal_code`. The schema would then be:

```
api:
  mapping:
    rate:
      mapping:
        origin_postal_code
```